docs/src/getting_started.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Getting Started Guide
+slug: /getting_started
+---
+
+# Getting Started with Charmil
+
+This guide assumes that you already have a working Go environment, if not then please refer to
+[this page](https://golang.org/doc/install) first.
+
+## Charmil SDK
+
+This is the component responsible for constructing modular, multi-repo Golang CLIs with Cobra while providing a variety of high-level solutions for typical challenges like configuration, internationalization, etc.
+&nbsp;
+
+- ### Installation:
+
+  First use `go get` to install the latest tagged release of the framework.
+  The following command will install the framework along with its dependencies:
+
+  ```bash
+  $ go get -u github.com/aerogear/charmil
+  ```
+
+  Next, import the required package into your project:
+
+  - For the config management package:
+
+    ```go
+    import "github.com/aerogear/charmil/core/config"
+    ```
+
+  - For utils packages (eg. color, localize, logging, etc.):
+
+    ```go
+    import "github.com/aerogear/charmil/core/utils/{name_of_the_required_package}"
+    ```
+
+- ### Usage:
+
+  For usage instructions on Charmil SDK, please refer to the docs [over here](./).
+
+## Charmil Validator
+
+- ### Installation:
+
+  First use `go get` to install the latest tagged release of the framework.
+  The following command will install the framework along with its dependencies:
+
+  ```bash
+  $ go get -u github.com/aerogear/charmil
+  ```
+
+  Next, import the Charmil Validator package into your project:
+
+  ```go
+  import "github.com/aerogear/charmil/validator"
+  ```
+
+- ### Usage:
+
+  For usage instructions on Charmil Validator, please refer to [this link](./charmil_validator.md#how-to-use).
+
+## Charmil Starter
+
+The Charmil Starter Template allows users to quickly create plugins and host commands that make use of the features offered by Charmil. This template includes all of the tools and solutions needed to create a command-line utility.
+&nbsp;
+
+- ### Usage:
+
+  The Charmil Starter Template can be used by any of the following ways:
+
+  - By cloning the [Charmil Starter Github repository](https://github.com/aerogear/charmil-starter)
+    ```bash
+    $ git clone https://github.com/aerogear/charmil-starter.git
+    ```
+  - By using the `charmil init` command offered by the Charmil CLI
+    ```bash
+    $ charmil init
+    ```
+
+## Charmil CLI
```

Comment:
Again. We using CLI before it is being presented. I think this should be first then starter then rest (possibly separate chapters/md files

Reply:
Resolved via commit: https://github.com/aerogear/charmil/pull/241/commits/6aa5fef405df8862744aca380661e39f78b83067